plugins/gopacket/run.go

```
@@ -0,0 +1,247 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	pathAndQuery := plugins.GetPathWithQuery(request.URL)
+
+	destinationNamespace := a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ServerIP)
+
+	requestID :=
+
+	return &models.Telemetry{
+		DestinationAddress:   item.ConnectionInfo.ServerIP + ":" + item.ConnectionInfo.ServerPort,
+		DestinationNamespace: destinationNamespace, // TODO if this is empty, it is a request to external. but in apiclarity we assume that empty namespace means internal
+		Request: &models.Request{
+			Common: &models.Common{
+				TruncatedBody: truncatedBodyReq,
+				Body:          reqBody,
+				Headers:       plugins.CreateHeaders(request.Header),
+				Version:       item.Protocol.Version,
+			},
+			Host:   request.Host,
+			Method: request.Method,
+			Path:   pathAndQuery,
+		},
+		RequestID: "",
+		Response: &models.Response{
+			Common: &models.Common{
+				TruncatedBody: truncatedBodyRes,
+				Body:          resBody,
+				Headers:       plugins.CreateHeaders(response.Header),
+				Version:       item.Protocol.Version,
+			},
+			StatusCode: strconv.Itoa(response.StatusCode),
+		},
+		Scheme:        item.Protocol.Name,
+		SourceAddress: item.ConnectionInfo.ClientIP + ":" + item.ConnectionInfo.ClientPort,
+	}, nil
+}
+
+func loadExtensions() ([]*api.Extension, error) {
+	var extensions []*api.Extension
+	files, err := ioutil.ReadDir("/app/extensions")
+	if err != nil {
+		return nil, fmt.Errorf("failed to read dir: %v", err)
+	}
+	extensions = make([]*api.Extension, len(files))
+	for i, file := range files {
+		filename := file.Name()
+		extension := &api.Extension{
+			Path: path.Join("/app/extensions", filename),
+		}
+		plug, err := plugin.Open("/app/extensions/http.so")
+		if err != nil {
+			return nil, fmt.Errorf("failed to open plugin: %v", err)
+		}
+		extension.Plug = plug
+		symDissector, err := plug.Lookup("Dissector")
```

error is not checked

---

It is checked in line 228:
```
if err != nil || !ok {
			return nil, fmt.Errorf("failed to load the extension: %s", extension.Path)
		}
```

---

not a lovely thing to do